cardano-cli/src/Cardano/CLI/Shelley/Parsers.hs

```diff
@@ -609,11 +609,28 @@ pTransaction =
     Opt.subparser
       $ Opt.command "sign-witness" assembleInfo <> Opt.internal
 
+  pScriptValidity :: Parser ScriptValidity
+  pScriptValidity = asum
+    [ Opt.flag' ScriptValid $ mconcat
+      [ Opt.long "script-valid"
+      , Opt.help "Assertion that the script is valid. (default)"
+      ]
+    , Opt.flag' ScriptInvalid $ mconcat
+      [ Opt.long "script-invalid"
+      , Opt.help $ mconcat
+        [ "Assertion that the script is invalid.  "
+        , "If a transaction is submitted with such a script, "
+        , "the script will fail and the collateral taken"
```

Comment:
We should have it default to `ScriptValid` as users will want to know if their script validates without losing their collateral.

Reply:
It defaults to `TxScriptValidityNone`, which ultimately means `ScriptValid` here: https://github.com/input-output-hk/cardano-node/blob/1222dba100d0cd49eaa2634cc9d1380023185577/cardano-api/src/Cardano/Api/TxBody.hs#L292

Reply:
That seems semantically incorrect. `TxScriptValidityNone` is used within the context of not being available for a given era.